cpplint.py

```diff
@@ -4548,8 +4559,16 @@ def _DropCommonSuffixes(filename):
   Returns:
     The filename with the common suffix removed.
   """
-  for suffix in ('test.cc', 'regtest.cc', 'unittest.cc',
```

Comment:
IIUC, you're rewriting this simple `for suffix` with `for bad_suffixes` that you fill up, because you need to generate it based on combinations of possible header/source extensions, and test file suffixes.

if this is the case, I think it would be more readable to just "say that":

    for suffix in itertools.chain(
        ('%s.%s' % (test_suffix.lstrip('_'), ext)
         for test_suffix, ext in itertools.product(_test_suffixes, _nonheader_extensions)),
        ('%s.%s' % (suffix, ext)
         for suffix, ext in itertools.product(['inl', 'imp', 'internal'], _header_extensions)))